File: checkPalindrome.cpp
class Solution{
  public:
  
  
    int isRev(int n,int temp){
        
        if(n==0){
            return temp;
        }
        
        temp=(temp*10)+(n%10);
        
        return isRev(n/10,temp);
        
    }
    bool isPalin(int N)
    {
        //Your code here
        //You may use a helper function if you like
        
        int temp=isRev(N,0);
        
        if(temp==N){
            return true;
        }
        
        else{
            return false;
        }
    }
};

File: checkPalindrome.py

def Reverse(self,N,k):
        
        if N==0:
            return k
        
        k=(k*10) + (N%10)
        
        return self.Reverse(N//10,k)
    def isPalin(self,N):
        #code here
        
        
        rev=self.Reverse(N,0);
        
        if rev==N:
            return 1
        else:
            return 0
            